src/opm/output/eclipse/AggregateNetworkData.cpp

```diff
@@ -0,0 +1,752 @@
+            }
+            ind_br = next_branch(n1, inlets, outlets);
+        }
+        n1 += 1;
+    }
+    return newInobr;
+}
+
+bool fixedPressureNode(const Opm::Schedule&         sched,
+                       const std::string&           nodeName,
+                       const size_t                 lookup_step
+                      )
+{
+    auto& network = sched[lookup_step].network();
+    bool fpn = (network.node(nodeName).terminal_pressure().has_value()) ? true  : false;
+    return fpn;
+}
+
+double nodePressure(const Opm::Schedule&               sched,
+                    const ::Opm::SummaryState&         smry,
+                    const std::string&                 nodeName,
+                    const Opm::UnitSystem&             units,
+                    const size_t                       lookup_step
+                   )
+{
+    using M = ::Opm::UnitSystem::measure;
+    double node_pres = 1.;
+    bool node_wgroup = false;
+    const auto& wells = sched.getWells(lookup_step);
+    auto& network = sched[lookup_step].network();
+
+    // If a node is a well group, set the node pressure to the well's thp-limit if this is larger than the default value (1.)
+    for (const auto& well : wells) {
+        const auto& wgroup_name = well.groupName();
+        if (wgroup_name == nodeName) {
+            if (well.isProducer()) {
+                const auto& pc = well.productionControls(smry);
+                if (pc.thp_limit >= node_pres) {
+                    node_pres = units.from_si(M::pressure, pc.thp_limit);
+                    node_wgroup = true;
+                }
+            }
+        }
+    }
+
+    // for nodes that are not well groups, set the node pressure to the fixed pressure potentially higher in the node tree
+    if (!node_wgroup) {
+        if (fixedPressureNode(sched, nodeName, lookup_step)) {
+            // node is a fixed pressure node
+            node_pres = units.from_si(M::pressure, network.node(nodeName).terminal_pressure().value());
+        }
+        else {
+            // find fixed pressure higher in the node tree
+            bool fp_flag = false;
+            auto node_name = nodeName;
+            auto upt_br = network.uptree_branch(node_name).value();
+            while (!fp_flag) {
+                if (fixedPressureNode(sched, upt_br.uptree_node(), lookup_step)) {
+                    node_pres = units.from_si(M::pressure, network.node(upt_br.uptree_node()).terminal_pressure().value());
+                    fp_flag = true;
+                } else {
+                    node_name = upt_br.uptree_node();
+                    if (network.uptree_branch(node_name).has_value()) {
+                        upt_br = network.uptree_branch(node_name).value();
+                    } else {
+                        std::stringstream str;
+                        str << "Node: " << nodeName << " has no uptree node with fixed pressure condition, uppermost node:  " << node_name;
```

Comment:
```C++
#include <fmt/format.h>
....
msg = fmt::format("Node: {0} has no uptree ....  node: {0}", node_name);
```

Reply:
OK - done. (Note: I used a copy of code used in other parts of opm-common - are all throw statements to be rewritten? - now there are different implementations in use)

Reply:
> OK - done

Good!

>are all throw statements to be rewritten? 

To rewrite existing statements is maybe a bit over the top - but new code should use fmtlib. Personally I also update old code to use fmtlib "when I am there".